### PR TITLE
fix(Scripts/Misc): Replace some AddItem/CreateItem with spells

### PR DESCRIPTION
**Pull request by Kitzunu**
* cherry-pick commit (https://github.com/TrinityCore/TrinityCore/commit/ed38206031f25fec73282044293238c9573a89d9)

Co-Authored-By: offl <11556157+offl@users.noreply.github.com>

## Changes Proposed:
-  Replace some AddItem/CreateItem with spells

## Tests Performed:

- none

## How to Test the Changes:

1. Check if you get rocket pack from Gunship.
2. Check if you get items from Bag of Heart Candies
3. Check if you get items from Mingo's Fortune Giblets
4. See if you get Arcane Disruptor from Chromie in Culling of Stratholme

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by temperrr** [comment]
There appears to be merge conflicts.

**Comment by FlamePrince** [comment]
> Thanks for your PR! There appears to be merge conflicts, could you take a look and set the labels back when fixed?

Hello @Kitzunu 

As an addendum, I started my testing with this very PR and LET ME TELL YOU, it was the most informative (albeit frustrating) experience. A couple of findings while I was attempting to test this PR:

- https://wowgaming.altervista.org/aowow/?item=43572 **also** does not work as intended when eaten, _but_ https://wowgaming.altervista.org/aowow/?spell=58886 works.
- Bag of Heart candies give me candy hearts as intended, and the spell works as intended - this is on master. Just requires "Love is in the Air"/8 to be active.
- I attempted every variety of spell related to Mingo's Fortune and found none of them to work.

**Comment by Kitzunu** [comment]
I will look at it later. Ty

File: src/server/scripts/Events/love_in_air.cpp [hunk 1]
```diff
@@ -670,16 +670,19 @@ class spell_love_in_air_periodic_perfumes : public SpellScriptLoader
...
+    SPELL_CREATE_HEART_CANDY_2     = 26670,
+    SPELL_CREATE_HEART_CANDY_3     = 26671,
+    SPELL_CREATE_HEART_CANDY_4     = 26672,
+    SPELL_CREATE_HEART_CANDY_5     = 26673,
+    SPELL_CREATE_HEART_CANDY_6     = 26674,
+    SPELL_CREATE_HEART_CANDY_7     = 26675,
+    SPELL_CREATE_HEART_CANDY_8     = 26676
 };
 
+std::array<uint32, 8> const CreateHeartCandySpells = {SPELL_CREATE_HEART_CANDY_1, SPELL_CREATE_HEART_CANDY_2, SPELL_CREATE_HEART_CANDY_3, SPELL_CREATE_HEART_CANDY_4, SPELL_CREATE_HEART_CANDY_5, SPELL_CREATE_HEART_CANDY_6, SPELL_CREATE_HEART_CANDY_7, SPELL_CREATE_HEART_CANDY_8};
```
Comment (Winfidonarleyan) [inline src/server/scripts/Events/love_in_air.cpp:683]: `constexpr`

File: src/server/scripts/Events/love_in_air.cpp [hunk 2]
```diff
@@ -689,17 +692,20 @@ class spell_item_create_heart_candy : public SpellScriptLoader
     {
         PrepareSpellScript(spell_item_create_heart_candy_SpellScript);
 
+
```
Comment (Winfidonarleyan) [inline src/server/scripts/Events/love_in_air.cpp:695]:
```suggestion
```

File: src/server/scripts/Kalimdor/CavernsOfTime/CullingOfStratholme/culling_of_stratholme.cpp [hunk 3]
```diff
@@ -1404,51 +1404,64 @@ enum chromie
...
+        bool OnGossipSelect(Player* player, Creature* creature, uint32 /*sender*/, uint32 /*action*/)
         {
-            if (InstanceScript* pInstance = creature->GetInstanceScript())
+            // final menu id, show crates if hidden and add item if missing
+            if (player->PlayerTalkClass->GetGossipMenu().GetMenuId() == 9595)
             {
-                if (pInstance->GetData(DATA_ARTHAS_EVENT) == COS_PROGRESS_NOT_STARTED)
+                if (InstanceScript* pInstance = creature->GetInstanceScript())
+                    if (pInstance->GetData(DATA_ARTHAS_EVENT) == COS_PROGRESS_NOT_STARTED)
+                        pInstance->SetData(DATA_SHOW_CRATES, 1);
```
Comment (Winfidonarleyan) [inline src/server/scripts/Kalimdor/CavernsOfTime/CullingOfStratholme/culling_of_stratholme.cpp:1435]: brrr

File: src/server/scripts/Kalimdor/CavernsOfTime/CullingOfStratholme/culling_of_stratholme.cpp [hunk 4]
```diff
@@ -1404,51 +1404,64 @@ enum chromie
...
+            // Skip Event
+            else if (player->PlayerTalkClass->GetGossipMenu().GetMenuId() == 11277)
+            {
+                if (InstanceScript* pInstance = creature->GetInstanceScript())
+                {
+                    if (pInstance->GetData(DATA_ARTHAS_EVENT) == COS_PROGRESS_NOT_STARTED)
+                    {
+                        pInstance->SetData(DATA_ARTHAS_EVENT, COS_PROGRESS_FINISHED_INTRO);
+                        if (Creature* arthas = ObjectAccessor::GetCreature(*creature, pInstance->GetGuidData(DATA_ARTHAS)))
+                            arthas->AI()->Reset();
```
Comment (Winfidonarleyan) [inline src/server/scripts/Kalimdor/CavernsOfTime/CullingOfStratholme/culling_of_stratholme.cpp:1451]: brr

**Comment by FrancescoBorzi** [comment]
what's the status of this?

**Comment by Kitzunu** [comment]
todo

**Comment by Kitzunu** [comment]
> https://wowgaming.altervista.org/aowow/?item=43572 also does not work as intended when eaten, but https://wowgaming.altervista.org/aowow/?spell=58886 works.

I don't touch these spells.

**Comment by Kitzunu** [comment]
Ok now I've fixed it all

**Comment by Kitzunu** [comment]
cc @FlamePrince

**Comment by Nefertumm** [comment]
Tested all the spells with .cast command, except for the icc one. All are working.

**Review by Nyeriah — APPROVED** [review]

**Pull request merged**